---
title: "hejpakken"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{hejpakken}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(hejpakken)
```

## Description

## Functions of hejpakken

In hejpakken there are 5 builtin functions:

1.  Nucleotide_sequence_sampler

2.  Amino_acid_plot

### Nucleotide sequence sampler

The nucleotide sequence sampler samples a selected number of DNA bases into a DNA sequence.

As an example, here is a sequence of 24 bases:

```{r}
example_sequence_24 <- nucleotide_sequence_sampler(24)
print(example_sequence_24)
```

### Amino acid frequency plot

The amino acid plot illustrates the count of each amino acids within an amino acid sequence in a barchart

The amino acid count of insulin is illustrated with the function below:

```{r}
INS <- "MALWMRLLPLLALLALWGPDPAAAFVNQHLCGSHLVEALYLVCGERGFFYTPKTRREAEDLQVGQVELGGGPGAGSLQPLALEGSLQKRGIVEQCCTSICSLYQLENYCN"

amino_acid_plot(INS)
```

### Merging all the functions

The functions can be used all together, where you can illustrate the frequency of different amino acids, based on random base sampling. This can illustrate that if the base is sampled randomly, then some amino acids are of higher likability than others.

```{r}
#150 random bases are sampled
dna_bases <- nucleotide_sequence_sampler(150)
#Bases are converted to RNA string
rna_bases <- convert_T_to_U(dna_bases)
#The rna bases are converted into codons
codons <- extract_codons(rna_bases)
#The codons are translated into an amino acid sequence
aa_sequence <- codon_to_AA(codons)
#The amino acids are illustrated in a count barplot
plot <- amino_acid_plot(aa_sequence)

plot
```

## Further development of hejpakken
